### PR TITLE
Spike: Update proposal submit flow

### LINKED ISSUE #15: Click "Cancel" on the CENNZnet Wallet popup doesn't close the progress dialog

### LINKED ISSUE #16: Rename component `WalletSelect` to `AccountSelect`

### PR DESCRIPTION
**Pull request by ken-futureverse**
## Summary

Make several updates to the codebase, mostly around the `/propsal/new` page

## Changes

- Rename path from `proposals` to `proposal`
- Add a fix to make two of the `prettier` plugins work together
- Use `getStaticProps` to extract the extrinsic metadata instead of relying on a static json file
- Apply new layout for the New Proposal form
- Add a backdrop `Layout` component to lighten up the page
- Fix the issue with the Progress Dialog (closes #15)
- Rename `WalletSelect` with `AccountSelect` (closes #16)

## Notes

@aidan-starke this PR is still missing the actual submission of the form, but feel like it would be beneficial to be merged as it contains a few quality of life fixes. Let me know if you want me to run through this with you.

Screen Shot 2022-07-07 at 12 03 29

### CODE REVIEW

File: apps/gov-website/pages/api/proposal/[pid].ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import mongoose from "mongoose";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { MONGODB_SERVER } from "@app-gov/service/constants";
+
+export default async function handler(
+	req: NextApiRequest,
+	res: NextApiResponse
+) {
+	try {
```
Comment (aidan-starke) [inline apps/gov-website/pages/api/proposal/[pid].ts]: This file can be removed

File: libs/web/components/src/MarkdownField.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+> = ({ className, children, value, rows = 10, ...props }) => {
+	const [preview, setPreview] = useState<boolean>(false);
+	const onWriteClick = useCallback(() => {
+		setPreview(false);
+	}, []);
+
+	const onPreviewClick = useCallback(() => {
+		setPreview(true);
+	}, []);
+
```
Comment (aidan-starke) [inline libs/web/components/src/MarkdownField.tsx:22]: why use callbacks over inline here?

File: libs/service/cennznet/src/extractCallableExtrinsics.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+interface CallableExtrinsics {
+	[section: string]: Extrinsic;
+}
+
+export const extractCallableExtrinsics = async (
+	api: Api
+): Promise<CallableExtrinsics> => {
+	const sections = Object.keys(api.tx)
```
Comment (aidan-starke) [inline libs/service/cennznet/src/extractCallableExtrinsics.ts:19]: this is dope

**Review by aidan-starke — DISMISSED** [review]

File: apps/gov-website/pages/api/proposal/[pid].ts (hunk 1 above)
Comment (aidan-starke) [reply apps/gov-website/pages/api/proposal/[pid].ts]: might need to rebase from develop? it was removed in my last PR

File: libs/web/components/src/MarkdownField.tsx (hunk 2 above)
Comment (ken-futureverse) [reply libs/web/components/src/MarkdownField.tsx:22]: just OCD really, inline creates new function every time, meaning the component receiving that prop will  be also re-render even though it technically doesn't change anything

File: libs/service/cennznet/src/extractCallableExtrinsics.ts (hunk 3 above)
Comment (ken-futureverse) [reply libs/service/cennznet/src/extractCallableExtrinsics.ts:19]: ✌

**Review by aidan-starke — DISMISSED** [review]

**Review by aidan-starke — APPROVED** [review]

**Pull request merged**